django/db/models/fields/__init__.py

```diff
@@ -175,6 +173,22 @@ def __init__(self, verbose_name=None, name=None, primary_key=False,
         self._error_messages = error_messages  # Store for deconstruction later
         self.error_messages = messages
 
+    def _get_choices(self):
+        return self._choices
+
+    def _set_choices(self, value):
+        if isinstance(value, (collections.abc.Iterator, collections.abc.Mapping)):
+            if isinstance(value, collections.abc.Mapping):
+                value = value.items()
+            # Convert potentially nested dictionaries to a flat list of tuples structure.
+            value = [
+                (k, list(v.items()) if isinstance(v, collections.abc.Mapping) else v)
+                for k, v in value
+            ]
+        self._choices = value
+
+    choices = property(_get_choices, _set_choices)
```

Comment:
better to use the `@property` syntax now, no?

Reply:
Good idea, fixed 👍